packages/react/src/transactions/useUnfollow.ts

```diff
@@ -1,61 +1,70 @@
-import { ProfileFragment } from '@lens-protocol/api-bindings';
+import {
+  isFollowTransactionFor,
+  isUnfollowTransactionFor,
+  ProfileFragment,
+  ProfileOwnedByMeFragment,
+  useHasPendingTransaction,
+  useWaitUntilTransactionSettled,
+} from '@lens-protocol/api-bindings';
 import {
   PendingSigningRequestError,
   TransactionKind,
   UserRejectedError,
   WalletConnectionError,
 } from '@lens-protocol/domain/entities';
-import { UnfollowRequest } from '@lens-protocol/domain/use-cases/profile';
-import { useState } from 'react';
+import { failure, PromiseResult } from '@lens-protocol/shared-kernel';
 
-import { TransactionState, useHasPendingTransaction } from './adapters/TransactionQueuePresenter';
+import { Operation, useOperation } from '../helpers';
 import { useUnfollowController } from './adapters/useUnfollowController';
 
 export class PrematureUnfollowError extends Error {
-  constructor(message: string) {
-    super(message);
-  }
+  name = 'PrematureUnfollowError' as const;
 }
 
 export type UseUnfollowArgs = {
-  profile: ProfileFragment;
+  followee: ProfileFragment;
+  follower: ProfileOwnedByMeFragment;
 };
 
-export function useUnfollow({ profile }: UseUnfollowArgs) {
-  const [error, setError] = useState<
-    | PendingSigningRequestError
-    | WalletConnectionError
-    | UserRejectedError
-    | PrematureUnfollowError
-    | null
-  >(null);
+export type UnfollowOperation = Operation<
+  void,
+  PendingSigningRequestError | WalletConnectionError | UserRejectedError | PrematureUnfollowError
+>;
+
+export function useUnfollow({ followee, follower }: UseUnfollowArgs): UnfollowOperation {
   const unfollow = useUnfollowController();
 
-  const hasPendingTx = useHasPendingTransaction(
-    (transaction): transaction is TransactionState<UnfollowRequest> =>
-      transaction.request.kind === TransactionKind.UNFOLLOW_PROFILE &&
-      transaction.request.profileId === profile.id,
+  const hasPendingFollowTx = useHasPendingTransaction(
+    isFollowTransactionFor({ profileId: followee.id, followerAddress: follower.ownedBy }),
   );
+  const waitUntilTransactionSettled = useWaitUntilTransactionSettled();
 
-  return {
-    unfollow: async () => {
-      if (!profile.isFollowedByMe && profile.isOptimisticFollowedByMe) {
-        setError(
-          new PrematureUnfollowError(`Your follow request for ${profile.handle} is still pending.`),
+  return useOperation(
+    async (): PromiseResult<
+      void,
+      | PendingSigningRequestError
```

Comment:
nit: same thought here for an reusable union of potential errors as in `useFollow`